src/server/json_family.cc

```diff
@@ -157,6 +159,48 @@ OpResult<json> GetJson(const OpArgs& op_args, string_view key) {
   return decoder.get_result();
 }
 
+string ConvertToJsonPointer(const string& json_path) {
+  if (json_path.empty() || json_path[0] != '$') {
+    return "";
+  }
+
+  vector<string> parts;
+  bool invalid_format = false;
+  std::size_t i = 1;
+  while (i < json_path.size()) {
+    bool is_array = false;
+    bool is_object = false;
+    if (json_path[i] == '[' && json_path[i + 1] == '\'') {
```

Comment:
I prefer testing once for `if (json_path[i] == '[')` and inside perform another check instead of checking it twice.